tests/test_plugin_loader.py

```
@@ -1469,17 +2095,22 @@ def test_run_type_error_non_string_args(self):
             _run(["git", 123])  # type: ignore[list-item]
         self.assertIn("all command arguments must be strings", str(cm.exception))
 
-    def test_run_value_error_empty_args(self):
+    def test_run_value_error_shell_true(self):
+        """Test _run raises ValueError for shell=True."""
+        with self.assertRaises(ValueError) as cm:
+            _run(["git", "status"], shell=True)  # noqa: S604 - intentional for test
+        self.assertIn("shell=True is not allowed in _run", str(cm.exception))
+
+    @patch("subprocess.run")
+    def test_run_value_error_empty_args(self, mock_subprocess):
```

Ruff (0.14.4)

Unused method argument: `mock_subprocess`

(ARG002)